src/traits.rs

```diff
@@ -1012,7 +1036,7 @@ pub trait SystemExt: Sized + Debug + Default {
     /// let s = System::new();
     /// println!("kernel version: {:?}", s.kernel_version());
     /// ```
-    fn kernel_version(&self) -> Option<String>;
+    fn kernel_version(&self) -> Option<[usize; 3]>;
```

Comment:
I'm definitely not sure it'll work out for macOS and windows...

Reply:
its tested on linux

Reply:
we could make functions that only work on linux (`kernel_name()` & `kernel_version()`) as linux-only with a `#[cfg(target_os = "linux")]` and a `#[cfg(not(target_os = "linux"))]` and leave `kernel_release()` as the cross-platform one.

Reply:
Except that it's definitely not the aim of this crate. Code using this crate should compile on all supported platforms.

I don't want any platform-specific functionalities exposed to the users.

Reply:
We still can make those functions run on all platforms, we could provide a solution for those:
* `kernel_name()`:
  * in Windows, the kernel name would be `NT` or `MS-DOS`, and idk if even LLVM compiles to msdos so...
  * in macOS, the kernel is `darwin`
  * in other OSes, we can just remove all numbers from the start and the end of `kernel_release()`
* `kernel_version`:
  * we could return a vector, so we don't have the array limitations that would make cross-platform impossible.
  * if there are numbers at the beginning of the kernel release get them and use the same strategy for pushing them to the vector, and fallback to the end of the string if there are no numbers at the beginning.